engine/runtime-with-instruments/src/test/scala/org/enso/interpreter/test/instrument/RuntimeServerTest.scala

```diff
@@ -2449,6 +2449,167 @@ class RuntimeServerTest
+    val requestId  = UUID.randomUUID()
+    val moduleName = "Enso_Test.Test.Main"
+    val newline    = System.lineSeparator()
+
+    context.send(Api.Request(requestId, Api.CreateContextRequest(contextId)))
+    context.receive shouldEqual Some(
+      Api.Response(requestId, Api.CreateContextResponse(contextId))
+    )
+
+    def template(text: String) =
+      s"""from Standard.Base.Data.Numbers import Number
+         |import Standard.Base.IO
+         |
+         |main = IO.println "${text}"
+         |""".stripMargin.linesIterator.mkString("\n")
+
+    val code = template("I'm a file!")
+
+    // Create a new file
+    val mainFile = context.writeMain(code)
+
+    // Open the new file
+    context.send(Api.Request(Api.OpenFileNotification(mainFile, code)))
+    context.receiveNone shouldEqual None
+    context.consumeOut shouldEqual List()
+
+    // Push new item on the stack to trigger the re-execution
+    context.send(
+      Api.Request(
+        requestId,
+        Api.PushContextRequest(
+          contextId,
+          Api.StackItem
+            .ExplicitCall(
+              Api.MethodPointer(moduleName, "Enso_Test.Test.Main", "main"),
+              None,
+              Vector()
+            )
+        )
+      )
+    )
+    context.receiveNIgnorePendingExpressionUpdates(
+      2
+    ) should contain theSameElementsAs Seq(
+      Api.Response(requestId, Api.PushContextResponse(contextId)),
+      context.executionComplete(contextId)
+    )
+    context.consumeOut shouldEqual List("I'm a file!")
+
+    /*
+      Modify the file:
+      """from Standard.Base.Data.Numbers import Number
+        |import Standard.Base.IO
+        |
+        |Number.lucky = 42
+        |
+        |main = IO.println "I'm a modified!"
+        |""".stripMargin.linesIterator.mkString("\n")
+     */
+    context.send(
+      Api.Request(
+        Api.EditFileNotification(
+          mainFile,
+          Seq(
+            TextEdit(
+              model.Range(model.Position(3, 25), model.Position(3, 29)),
+              "modified"
+            ),
+            TextEdit(
+              model.Range(model.Position(3, 0), model.Position(3, 0)),
+              s"Number.lucky = 42$newline$newline"
+            )
+          ),
+          execute = true
+        )
+      )
+    )
+    context.receiveNIgnoreExpressionUpdates(
+      1
+    ) should contain theSameElementsAs Seq(
+      context.executionComplete(contextId)
+    )
+    context.consumeOut shouldEqual List("I'm a modified!")
+
+    // Simulate file update in FS
+    val prompt = "I'm a foo"
+    context.writeMain(template(prompt))
+
+    context.send(Api.Request(requestId, Api.PopContextRequest(contextId)))
+    context.receive shouldEqual Some(
+      Api.Response(requestId, Api.PopContextResponse(contextId))
+    )
+    // Push new item on the stack to trigger the re-execution
+    context.send(
+      Api.Request(
+        requestId,
+        Api.PushContextRequest(
```

Comment:
btw, there is `Api.RecomputeContextRequest` if you just need to trigger the re-execution

Reply:
Thanks!